classes/log/EmailLogDAO.inc.php

```diff
@@ -64,14 +64,15 @@ function _getByEventType($assocType, $assocId, $eventType, $userId = null, $rang
 				(int) $eventType);
 		if ($userId) $params[] = $userId;
 
+		$isSqlServer = Config::getVar('database', 'ms_sql');
 		$result = $this->retrieveRange(
 			'SELECT	e.*
 			FROM	email_log e' .
 			($userId ? ' LEFT JOIN email_log_users u ON e.log_id = u.email_log_id' : '') .
 			' WHERE	e.assoc_type = ? AND
 				e.assoc_id = ? AND
-				e.event_type = ?' .
-				($userId ? ' AND u.user_id = ?' : ''),
+				e.event_type = ?'
+				. ($userId && !$isSqlServer ? ' AND u.user_id = ?' : ''),
```

Comment:
Won't this cause the function to return matches from other users? Is there a reason SQL Server won't accept the condition?

Reply:
You are right on this one, we should not exclude the condition. Sadly, there is a major problem we can't totally explain : the "email_log_users" table is totally empty. This said, SQL server can't return any results with the user ID filter.

The problem seems located in the method "_insertLogUserIds" when the following pattern is used :

```
$pattern = '/(?<=\<)[^\>]*(?=\>)/';
preg_match_all($pattern, $recipients, $matches);
```

Why the method is expecting emails formed this way : <EMAIL>? In our case, the characters < and > are not present in any conditions within the recipients.

Reply:
@nibou230, I suspect it's expecting emails to be provided in the form `Screen Name <emailaddress@domain.com>`.